### PR TITLE
Add ability to use a virtual list in combobox

### PR DESCRIPTION
**Pull request by jpzwarte**

### CODE REVIEW

File: packages/components/combobox/src/grouped-option.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { property } from 'lit/decorators.js';
+import styles from './grouped-option.scss.js';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'sl-combobox-grouped-option': GroupedOption;
+  }
+}
+
+export class GroupedOption extends Option {
```
Comment (anna-lach) [inline packages/components/combobox/src/grouped-option.ts:12]: Can you please also add a small description above just like we have in other components? :)

File: packages/components/combobox/src/combobox.ts [hunk 2]
```diff
@@ -77,8 +94,12 @@ export class Combobox<T = unknown> extends FormControlMixin(ScopedElementsMixin(
   /** Message element for when filtering results did not yield any results. */
   #noMatch?: NoMatch;
 
-  /** Monitor the DOM for new options. */
-  #observer = new MutationObserver(() => this.#updateOptions());
+  /** Update the width of the popover while open. */
+  #observer = new ResizeObserver(entries => {
+    if (this.wrapper?.matches(':popover-open')) {
```
Comment (anna-lach) [inline packages/components/combobox/src/combobox.ts:99]: Or maybe we can use `isPopoverOpen` here?

File: packages/components/combobox/src/combobox.ts [hunk 3]
```diff
@@ -116,16 +140,13 @@ export class Combobox<T = unknown> extends FormControlMixin(ScopedElementsMixin(
   @event({ name: 'sl-blur' }) blurEvent!: EventEmitter<SlBlurEvent>;
 
   /** @internal Emits when the value changes. */
-  @event({ name: 'sl-change' }) changeEvent!: EventEmitter<SlChangeEvent<T | T[] | undefined>>;
+  @event({ name: 'sl-change' }) changeEvent!: EventEmitter<SlChangeEvent<U | U[] | undefined>>;
 
   /** @internal The create custom option option (used when `allowCustomValues` is set). */
```
Comment (anna-lach) [inline packages/components/combobox/src/combobox.ts:145]: Small typo `option option` :)

File: packages/components/combobox/src/combobox.ts [hunk 4]
```diff
@@ -215,8 +260,53 @@ export class Combobox<T = unknown> extends FormControlMixin(ScopedElementsMixin(
   override willUpdate(changes: PropertyValues<this>): void {
     super.willUpdate(changes);
 
-    if (changes.has('options') || changes.has('value')) {
-      this.#updateSelected();
+    if (changes.has('multiple')) {
+      if (this.multiple) {
```
Comment (anna-lach) [inline packages/components/combobox/src/combobox.ts:264]: Maybe
`this.listbox?.toggleAttribute('aria-multiselectable', this.multiple);
`
would be simpler?

File: packages/components/listbox/src/option-group-header.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import { property } from 'lit/decorators.js';
+import styles from './option-group-header.scss.js';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'sl-option-group-header': OptionGroupHeader;
+  }
+}
+
+export class OptionGroupHeader extends LitElement {
```
Comment (anna-lach) [inline packages/components/listbox/src/option-group-header.ts:11]: Can you please add a short description of the component above?

**Review by anna-lach — APPROVED** [review]

File: packages/components/combobox/src/combobox.ts (hunk 4 above)
Comment (jpzwarte) [reply packages/components/combobox/src/combobox.ts:264]: No, that would make it a `boolean` and it needs to be a `string`

**Pull request merged**